dask_expr/tests/test_resample.py

```diff
@@ -20,15 +20,56 @@ def df(pdf):
     yield from_pandas(pdf, npartitions=4)
 
 
-def test_resample_count(df, pdf):
-    result = df.resample("2T").count()
-    expected = pdf.resample("2T").count()
+@pytest.mark.parametrize(
+    "api",
+    [
+        "count",
+        "prod",
+        "mean",
+        "sum",
+        "min",
+        "max",
+        "first",
+        "last",
+        "var",
+        "std",
+        "size",
+        "nunique",
+        "median",
+        "quantile",
+        "ohlc",
+        "sem",
+    ],
+)
+def test_resample_apis(df, pdf, api):
+    result = getattr(df.resample("2T"), api)()
+    expected = getattr(pdf.resample("2T"), api)()
     assert_eq(result, expected)
 
-    result = df.resample("2T").count()["foo"]
-    expected = pdf.resample("2T").count()["foo"]
+    # No column output
+    if api not in ("size",):
+        result = getattr(df.resample("2T"), api)()["foo"]
+        expected = getattr(pdf.resample("2T"), api)()["foo"]
+        assert_eq(result, expected)
+
+        q = result.simplify()
+        eq = getattr(df["foo"].resample("2T"), api)().simplify()
+        assert q._name == eq._name
+
+
+def test_resample_agg(df, pdf):
+    def my_sum(vals, foo=None, *, bar=None):
+        return vals.sum()
+
+    # TODO: w/o .compute() assert_eq fails w/ dtype assertion error
```

Comment:
Can you post a traceback?

Reply:
```
ddf = <dask_expr.expr.Index: expr=Index(frame=ResampleAgg(frame=df, rule=2T, kwargs={}, fill_value=nan, how_args=(<function test_resample_agg.<locals>.my_sum at 0x7f11fbdf4f70>, 'foo'), how_kwargs={'bar': 'bar'}))>
res = DatetimeIndex(['2000-01-01 00:00:00', '2000-01-01 00:02:00',
               '2000-01-01 00:04:00', '2000-01-01 00:06:00',
               '2000-01-01 00:08:00', '2000-01-01 00:10:00'],
              dtype='datetime64[ns]', freq='2T')
numeric_equal = True

    def assert_dask_dtypes(ddf, res, numeric_equal=True):
        """Check that the dask metadata matches the result.

        If `numeric_equal`, integer and floating dtypes compare equal. This is
        useful due to the implicit conversion of integer to floating upon
        encountering missingness, which is hard to infer statically."""

        eq_type_sets = [{"O", "S", "U", "a"}]  # treat object and strings alike
        if numeric_equal:
            eq_type_sets.append({"i", "f", "u"})

        def eq_dtypes(a, b):
            return any(
                a.kind in eq_types and b.kind in eq_types for eq_types in eq_type_sets
            ) or (a == b)

        if not is_dask_collection(res) and is_dataframe_like(res):
            for a, b in pd.concat([ddf._meta.dtypes, res.dtypes], axis=1).itertuples(
                index=False
            ):
                assert eq_dtypes(a, b)
        elif not is_dask_collection(res) and (is_index_like(res) or is_series_like(res)):
            a = ddf._meta.dtype
            b = res.dtype
>           assert eq_dtypes(a, b)
E           AssertionError

../../mambaforge/envs/dask-expr/lib/python3.10/site-packages/dask/dataframe/utils.py:699: AssertionError
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> entering PDB >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> PDB post_mortem >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
> /home/milesg/mambaforge/envs/dask-expr/lib/python3.10/site-packages/dask/dataframe/utils.py(699)assert_dask_dtypes()
-> assert eq_dtypes(a, b)
(Pdb) p a
dtype('O')
(Pdb) p b
dtype('<M8[ns]')
```

Reply:
Could you try:

```
getattr(meta_nonempty(self.frame._meta).resample(self.rule, **self.kwargs), self.how)()
```

as _meta for both classes? Maybe that keeps the dtype properly

Reply:
That's the ticket! Thanks! https://github.com/dask-contrib/dask-expr/pull/399/commits/42d4355dbc5f609f4cb73da93d90c2fc2610785e